tools/releases/version.sh

```diff
@@ -2,24 +2,28 @@
 
 # Kuma version is built as follows:
 # 1) If a git tag is present on the current commit, then the version is a git tag
-# 2) If the branch starts with "release" (like "release-1.3"), then the version is "$lastGitTag-$shortHash" (for example: 1.4.1-450174242)
-#    At the same time, if $lastGitTag is not final tag (for example 1.4.0-rc1 or 1.4.0-preview1), the version is "dev-$shortHash"
-# 3) If the branch does not start with "release", then the version is "dev-$shortHash" (for example: dev-450174242)
-
-lastGitTag=$(git describe --abbrev=0 --tags 2> /dev/null)
-shortHash=$(git rev-parse --short HEAD 2> /dev/null)
-currentBranch=$(git rev-parse --abbrev-ref HEAD 2> /dev/null)
+# 2) If the branch is "release-X.Y" look at the existing tags and use either X.Y.0-<shortHash> if there's none or
+#     the latest tag with a patch version increased by one (.e.g if latest tag is X.Y.1 the version will be `X.Y.2-<shortHash>`)
+# 3) In non release branch use `dev-$shortHash`
 
 # Note: this format must be changed carefully, other scripts depend on it
+exactTag=$(git describe --exact-match --tags > /dev/null 2>&1)
+if [[ ${exactTag} ]]; then # if we are on tag
+  echo "${exactTag}"
+  exit 0
+fi
 
-if git describe --exact-match --tags > /dev/null 2>&1; then # if we are on tag
-  echo "$lastGitTag"
+shortHash=$(git rev-parse --short HEAD 2> /dev/null)
+currentBranch=$(git rev-parse --abbrev-ref HEAD 2> /dev/null)
+if [[ ${currentBranch} == release-* ]]; then
+    releasePrefix=${currentBranch//release-/}
+    lastGitTag=$(git tag -l | grep -E "^${releasePrefix}\.[0-9]+$" | sort | tail -1)
```

Comment:
Hmm `sort` is lexicographic and not numeric, right?
```
$ echo "1.5\n1.10" | sort
1.10
1.5
$ echo "1.5\n1.9" | sort
1.5
1.9

```

Reply:
It will only matter if we have more than 10 patch versions (the major, minor will be the same) might be worth fixing this though to avoid future surprise

Reply:
TIL:

```
       -V, --version-sort
              natural sort of (version) numbers within text
```

Reply:
```
echo "1.5.1
1.2.11
1.5.9
1.6.0
1.5.10" | sort -V
1.2.11
1.5.1
1.5.9
1.5.10
1.6.0
```